src/Admin/SnapshotAdmin.php

```diff
@@ -30,13 +33,67 @@ final class SnapshotAdmin extends AbstractAdmin
 {
     protected $classnameLabel = 'Snapshot';
 
+    private TransformerInterface $transformer;
+
+    private PageManagerInterface $pageManager;
+
+    public function __construct(
+        TransformerInterface $transformer,
+        PageManagerInterface $pageManager
+    ) {
+        parent::__construct();
+
+        $this->transformer = $transformer;
+        $this->pageManager = $pageManager;
+    }
+
     protected function getAccessMapping(): array
     {
         return [
             'batchToggleEnabled' => AdminPermissionMap::PERMISSION_EDIT,
         ];
     }
 
+    protected function alterNewInstance(object $object): void
+    {
+        if (!$this->hasRequest()) {
+            return;
+        }
+
+        $pageId = $this->getRequest()->query->get('pageId');
+
+        if (null === $pageId) {
+            return;
+        }
+
+        $object->setPage($this->pageManager->find($pageId));
+    }
+
+    protected function prePersist(object $object): void
+    {
+        $page = $this->isChild() ? $this->getParent()->getSubject() : $object->getPage();
+        \assert(null === $page || $page instanceof PageInterface);
+
+        if (null === $page) {
+            return;
+        }
+
+        $snapshot = $this->transformer->create($page);
+
+        $object->setPage($snapshot->getPage());
+        $object->setUrl($snapshot->getUrl());
+        $object->setEnabled($snapshot->getEnabled());
+        $object->setRouteName($snapshot->getRouteName());
+        $object->setPageAlias($snapshot->getPageAlias());
+        $object->setType($snapshot->getType());
+        $object->setName($snapshot->getName());
+        $object->setPosition($snapshot->getPosition());
+        $object->setDecorate($snapshot->getDecorate());
+        $object->setSite($snapshot->getSite());
+        $object->setParentId($snapshot->getParentId());
+        $object->setContent($snapshot->getContent());
```

Comment:
I do not like this a lot, but not sure is a better way to do this... Have talked to @VincentLanglet already, but that's a missing feature from the AdminBundle. I can just do set on the already given object.

Reply:
i still think we might add a new function to Transformer to handle this

Reply:
Already answered about that, it will not solve the issue, just hide it somewhere else.

Reply:
The method create does
```
public function create(PageInterface $page): SnapshotInterface
    {
        $snapshot = $this->snapshotManager->create();
```

If we do 
```
public function create(PageInterface $page, ?SnapshotInterface snapshot): SnapshotInterface
    {
        $snapshot ??= $this->snapshotManager->create();
```
it avoid you to set everything.

(And then we should rename the method) ?

@jordisala1991

Reply:
I will give a try to modify the create method.

Reply:
It should be better now.